client/dashboard/sites/overview/site-card.tsx

```diff
@@ -102,40 +103,81 @@ function FieldTitle( { children }: { children: React.ReactNode } ) {
 	);
 }
 
-function PlanDetails( { site, currentPlan }: { site: Site; currentPlan: Plan } ) {
-	if ( ! site.plan || ! currentPlan ) {
+function PlanDetails( { site }: { site: Site } ) {
+	const { data: currentPlan } = useQuery( siteCurrentPlanQuery( site.slug ) );
+
+	if ( ! site.plan ) {
 		return null;
 	}
 
 	const {
 		plan: { product_name_short, is_free: isFree },
 	} = site;
-	const { expiry, id } = currentPlan;
+
 	return (
 		<VStack>
 			<FieldTitle>{ __( 'Plan' ) }</FieldTitle>
 			{ product_name_short && <Text>{ product_name_short }</Text> }
-			<Text>{ getPlanExpirationMessage( { isFree, expiry } ) }</Text>
-			{ id ? (
-				<Button href={ `/purchases/subscriptions/${ site.slug }/${ id }` } variant="link">
-					{ __( 'Manage subscription' ) }
-				</Button>
-			) : (
+			<Text>{ getPlanExpirationMessage( { isFree, currentPlan } ) }</Text>
+			{ isFree ? (
 				<Button href={ `/plans/${ site.slug }` } variant="link">
 					{ __( 'Upgrade' ) }
 				</Button>
+			) : (
+				<Button
+					// @ts-expect-error inert is not typed
+					inert={ ( ! currentPlan ).toString() }
+					href={ currentPlan ? `/purchases/subscriptions/${ site.slug }/${ currentPlan.id }` : '' }
+					variant="link"
+				>
+					{ currentPlan ? (
+						__( 'Manage subscription' )
+					) : (
+						// Show a blurred text while we wait for the plan ID.
+						<TextBlur>{ __( 'Manage subscription' ) }</TextBlur>
+					) }
+				</Button>
 			) }
 		</VStack>
 	);
 }
 
-function getPlanExpirationMessage( { isFree, expiry }: { isFree: boolean; expiry?: string } ) {
+function getPlanExpirationMessage( {
+	isFree,
+	currentPlan,
+}: {
+	isFree: boolean;
+	currentPlan?: Plan;
+} ) {
 	if ( isFree ) {
 		return __( 'No expiration date.' );
 	}
-	return (
-		expiry &&
-		/* translators: %s: date of plan's expiration date. Eg.  August 20, 2025 */
-		sprintf( __( 'Expires on %s.' ), dateI18n( 'F j, Y', expiry ) )
+
+	/* translators: %s: date of plan's expiration date. Eg.  August 20, 2025 */
+	const expiresString = __( 'Expires on <time>%s</time>.' );
+
+	// Show a blurred time while we wait for the current plan.
+	if ( ! currentPlan ) {
+		const iso = new Date().toISOString();
+		return (
+			<TextBlur>
+				{ createInterpolateElement( sprintf( expiresString, dateI18n( 'F j, Y', iso ) ), {
+					time: <time dateTime={ iso } />,
+				} ) }
+			</TextBlur>
+		);
+	}
+
+	// Some plans don't have an expiry date, and we don't want to show it at
+	// all.
+	if ( ! currentPlan.expiry ) {
+		return __( 'No expiration date.' );
```

Comment:
Let's return nothing here as per the comment above?

Reply:
OK, I was just going to suggest to do it separately, not sure if it's worth handling it now since it's very likely that the design will change for A4A sites.

Reply:
Ok, sure, let's do that. I think the new design has it in card as well, so in that case it would be fixed height and now something that could cause a shift.